From 7cad0ff082977b9a4b5d7c90f0ed21ad79bb5676 Mon Sep 17 00:00:00 2001
From: Colin Walters <walters@verbum.org>
Date: Wed, 14 Nov 2012 19:48:01 -0500
Subject: [PATCH] Fix 16bpp in llvmpipe

Taken from Fedora mesa.spec:

* Mon Apr 02 2012 Adam Jackson <ajax@redhat.com> 8.0.2-2
- mesa-8.0.1-fix-16bpp.patch: Taken from Fedora mesa.spec
---
 src/mesa/state_tracker/st_manager.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/mesa/state_tracker/st_manager.c b/src/mesa/state_tracker/st_manager.c
index 88b886d..72f3139 100644
--- a/src/mesa/state_tracker/st_manager.c
+++ b/src/mesa/state_tracker/st_manager.c
@@ -530,6 +530,9 @@ st_context_teximage(struct st_context_iface *stctxi,
       if (util_format_get_component_bits(internal_format,
                UTIL_FORMAT_COLORSPACE_RGB, 3) > 0)
          internalFormat = GL_RGBA;
+      else if (util_format_get_component_bits(internal_format,
+                 UTIL_FORMAT_COLORSPACE_RGB, 0) == 5)
+         internalFormat = GL_RGB5;
       else
          internalFormat = GL_RGB;
 
-- 
1.7.7.6

